pkg/asset/cluster/tfvars.go

```diff
@@ -327,9 +327,9 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		for i, w := range workers {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*machinev1beta1.AzureMachineProviderSpec)
 		}
+
 		client := aztypes.NewClient(session)
-		hyperVGeneration, err := client.GetHyperVGenerationVersion(context.TODO(), masterConfigs[0].VMSize,
-			masterConfigs[0].OSDisk.ManagedDisk.StorageAccountType, masterConfigs[0].Location)
+		hyperVGeneration, err := client.GetHyperVGenerationVersion(context.TODO(), masterConfigs[0].VMSize, masterConfigs[0].Location)
```

Comment:
This change now looks out of place in this PR. You're dropping one of the arguments from `GetHyperVGenerationVersion` but how is that related to the `vmNetworkType`?

Reply:
This is something I realized as I was working through the code. There are arguments passed that are not required. I was doing some clean up as I went through. I can revert this change and leave the cruft for someone else.

Reply:
You can at least put it in its own commit so the reason for the change is clear for someone looking at the commit history down the line.